src/mscorlib/src/System/Array.cs

```diff
@@ -1707,13 +1707,14 @@ public static void Reverse<T>(T[] array, int index, int length)
                 ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);
             Contract.EndContractBlock();
 
+            ref T p = ref Unsafe.As<byte, T>(ref array.GetRawSzArrayData());
```

Comment:
Is there a way to do something similar to replace TrySZReverse in the non-generic Reverse?
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Array.cs#L1656

Reply:
Replacing TrySZReverse with C# implementation would be more profound change. I will leave it for a different day..